pkg/suggestion/v1alpha2/parsing_util.py

```diff
@@ -0,0 +1,144 @@
+            new_lower = float(param.feasible_space.min)
+            new_upper = float(param.feasible_space.max)
+        elif param.parameter_type == api_pb2.INT:
+            new_lower = int(param.feasible_space.min)
+            new_upper = int(param.feasible_space.max)
+        elif param.parameter_type == api_pb2.DISCRETE:
+            discrete_values = [int(x) for x in param.feasible_space.list]
+            new_lower = min(discrete_values)
+            new_upper = max(discrete_values)
+            discrete_info.append(
+                {"name": param.name, "values": discrete_values})
+        elif param.parameter_type == api_pb2.CATEGORICAL:
+            num_feasible = len(param.feasible_space.list)
+            new_lower = [0 for _ in range(num_feasible)]
+            new_upper = [1 for _ in range(num_feasible)]
+            categorical_info.append({
+                "name": param.name,
+                "values": param.feasible_space.list,
+                "number": num_feasible,
+            })
+        if isinstance(new_lower, Iterable):  # handles categorical parameters
+            lower_bounds.extend(new_lower)
+            upper_bounds.extend(new_upper)
+            dim += len(new_lower)
+        else:  # handles ints, doubles, and discrete parameters
+            lower_bounds.append(new_lower)
+            upper_bounds.append(new_upper)
+            dim += 1
+    parsed_config = ParameterConfig(name_ids,
+                                    dim,
+                                    lower_bounds,
+                                    upper_bounds,
+                                    parameter_types,
+                                    names,
+                                    discrete_info,
+                                    categorical_info)
+    return parsed_config
+
+
+def parse_previous_observations(parameters_list, dim, name_id, types, categorical_info):
+    parsed_X = np.zeros(shape=(len(parameters_list), dim))
+    for row_idx, parameters in enumerate(parameters_list):
+        offset = 0
+        for p in parameters:
+            map_id = name_id[p.name]
+            if types[map_id] in [api_pb2.DOUBLE, api_pb2.INT,
+                                 api_pb2.DISCRETE]:
+                parsed_X[row_idx, offset] = float(p.value)
+                offset += 1
+            elif types[map_id] == api_pb2.CATEGORICAL:
+                for ci in categorical_info:
+                    if ci["name"] == p.name:
+                        value_num = ci["values"].index(p.value)
+                        parsed_X[row_idx, offset + value_num] = 1
+                        offset += ci["number"]
+    return parsed_X
+
+
+def parse_metric(y_train, goal):
+    """
+    Parse the metric to the dictionary
+    """
+    y_array = np.array(y_train, dtype=np.float64)
+    if goal == api_pb2.MINIMIZE:
+        y_array *= -1
+    return y_array
+
+
+def parse_x_next_vector(x_next, param_types, param_names, discrete_info, categorical_info):
```

Comment:
Nit - The names of these methods (and the class) feel somewhat weird. "Parse*" seems to suggest taking in a string and extracting the contents from it.